Modules/Core/Common/test/itkImageBaseGTest.cxx

```diff
@@ -29,6 +29,14 @@
 
 namespace
 {
+// Tells whether GetImageDimension() returns ImageDimension, for all of the specified values of VImageDimension.
+template <unsigned int... VImageDimension>
+constexpr bool GetImageDimension_returns_ImageDimension{ (
+  (itk::ImageBase<VImageDimension>::GetImageDimension() == VImageDimension) && ...) };
+
+static_assert(GetImageDimension_returns_ImageDimension<2, 3, 4>, "GetImageDimension() should return ImageDimension.");
```

Comment:
My first _variadic_ [variable template](https://en.cppreference.com/w/cpp/language/variable_template) 😃 (`GetImageDimension_returns_ImageDimension<...>` is a boolean constant whose value depends on its template arguments.)  Note that the parentheses in `(itk::ImageBase<VImageDimension>::GetImageDimension() == VImageDimension) && ...)` appear necessary, in order to correctly parse the [fold expression](https://en.cppreference.com/w/cpp/language/fold).